networkx/linalg/algebraicconnectivity.py

```diff
@@ -244,7 +244,7 @@ def P(x):
             W -= (W.T * X * X.T).T
             project(W)
             # Compute the diagonal of P * L * P as a Jacobi preconditioner.
-            D = L.diagonal()
+            D = L.diagonal().astype(float)
             D += 2. * (asarray(X) * asarray(W)).sum(axis=1)
```

Comment:
Can we avoid an explicit astype() call by combining these three lines into a single multiline assignment statement?

I'd prefer using the automatic coercion, and the single assignment statement seems more clear anyway.

Thanks for all this by the way -- :)